### PR TITLE
Generate both ecdsa, rsa key in testcloud

### PR DESCRIPTION
**Pull request by lukaszachy**
Also cloud-init modified to actually change root password.

In the future we can make key type part of the user configuration). Or should I do it in the PR?

### CODE REVIEW

**Comment by lukaszachy** [comment]
After some thinking I'm not so sure.
Motivation is that newer Fedora or Centos have disabled RSA key as ssh authentication method. ECDSA type works for them (and for reasonably old Fedora / RHEL / Centos). But tmt will diverge from ssh-keygen which generates RSA as a default (when -t is not used).
How can user tell tmt to use ssh-keygen default if tmt changes the default? Because : missing type -> use tmt default; specified type -> use that type. Should we dedicate some value for 'ssh-keygen default'?

File: tmt/steps/provision/testcloud.py [hunk 1]
```diff
@@ -126,6 +127,11 @@ def import_testcloud():
 DEFAULT_BOOT_TIMEOUT = 60      # seconds
 DEFAULT_CONNECT_TIMEOUT = 60   # seconds
 
+# Image guessing related variables
+KOJI_URL = 'https://kojipkgs.fedoraproject.org/compose'
```
Comment (psss) [inline tmt/steps/provision/testcloud.py:131]: This seems to be some old code leftover, please rebase on the latest master.

**Comment by psss** [comment]
> After some thinking I'm not so sure.

I believe the important question here is if this new default will break connecting to some older distros which do not support `ecdsa` keys. For those we need to have a way how to make it work as well. A dedicated key for the key type might be a way.

**Comment by psss** [comment]
@frantisekz, could you please have a look as well?

**Comment by frantisekz** [comment]
> @frantisekz, could you please have a look as well?

+1 from me. Changes seem reasonable, doesn't break anything that is currently supported by testcloud.

**Review by psss — APPROVED** [review]
Looks good to me as well. Rebased on the latest master.

**Pull request merged**